### ISSUE TITLE
In-Game: [HTML5] array_last() gives an exception when calling a script function inside a struct inside an array

### ISSUE THREAD

**Issue by KormexGit** [YoYoGames/GameMaker-Bugs#8049]
### Description
Day2 of HTML5 is unfathomably cursed, another game crash through very specific circumstances.  Defining the function as a method within the struct instead of in a script will fix it, accessing the array index directly instead of using array_last will fix it, and even storing the return of array_last in a local variable first and then running the function fixes it. It only gives an exception in the specific circumstances I have in the steps.

### Steps To Reproduce
In a script:
```
function noop() {}
```
In an object:
```
struct = {
	step : noop
}

array = [struct];
array_last(array).step();
``` 
The last line will give `Unhandled Exception - TypeError: yythis1.gmlstep is not a function`

### Which version of GameMaker are you reporting this issue for?
IDE v2024.1100.0.634 Runtime v2024.1100.0.658

### Which operating system(s) are you seeing the problem on?
Windows 10.0.22631.0

### Which platform(s) are you seeing the problem on?
HTML5

[24428a65-1e62-4da1-9d6d-8d1d4b12abfb](https://api.gamemaker.io/api/github/downloads/24428a65-1e62-4da1-9d6d-8d1d4b12abfb)

**Comment by rwkay** [comment]
Unable to replicate in the current internal version - I think this was fixed with an earlier commit for #7949

**Comment by sihammill** [comment]
Fix verified in  IDE v2024.1100.0.653  Runtime v2024.1100.0.679
